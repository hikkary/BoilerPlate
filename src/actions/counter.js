export const INCREMENT_COUNTER = 'INCREMENT_COUNTER'
export const DECREMENT_COUNTER = 'DECREMENT_COUNTER'

export const incrementCounter = house => ({
  type: 'INCREMENT_COUNTER',
  payload: house
})

export const decrementCounter = house => ({
  type: 'DECREMENT_COUNTER',
  payload: house
})
