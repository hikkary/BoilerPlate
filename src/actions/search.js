export const DISPLAY_SEARCH = 'DISPLAY_SEARCH'

export const displaySearch = search => ({
  type: 'DISPLAY_SEARCH',
  payload: search
})
