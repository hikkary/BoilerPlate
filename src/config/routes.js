import React from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import Login from '../containers/login'
import HomePage from '../containers/homePage'
import HouseCharacters from '../containers/houseCharacters'
import Ranking from '../containers/ranking'
import Options from '../containers/options'

export default class Routes extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Login} />
          <Route path="/home" component={HomePage} />
          <Route path="/house/:houseName" component={HouseCharacters} />
          <Route path="/ranking" component={Ranking} />
          <Route path="/options" component={Options} />
        </Switch>
      </Router>
    )
  }
}
