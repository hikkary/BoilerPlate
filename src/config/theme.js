export default {
  color: {
    black: '#222',
    white: '#fff',
    loginBackground: '#f2debd',
    bookSeparator: '#580303',
    primaryColor: '#7f0909',
    secondaryColor: '#940B0B',
    tertiaryColor: '#CB4B4B',
    gray: '#979797'
  }
}
