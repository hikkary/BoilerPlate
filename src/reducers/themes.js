import { CHANGE_THEME } from '../actions/themes'
import { themeLight } from '../config/themes'

const initialState = {
  currentTheme: themeLight
}

export default (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_THEME:
      return {
        ...state,
        currentTheme: action.theme
      }

    default:
      return state
  }
}
