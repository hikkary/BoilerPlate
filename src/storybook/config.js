import React from 'react'
import { addDecorator, configure } from '@storybook/react'
import { ThemeProvider } from 'styled-components'
import centered from '@storybook/addon-centered'
import '../../node_modules/react-vis/dist/style.css'

import theme from '../config/theme'

const ThemeDecorator = storyFn => (
  <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
)
addDecorator(ThemeDecorator)
addDecorator(centered)

function loadStories() {
  require('../components/input/story.js')
  require('../components/loginForm/story.js')
  require('../components/panel/story.js')
  require('../components/texts/story.js')
  require('../components/graphTest/story.js')
  // You can require as many stories as you need.
}

configure(loadStories, module)
