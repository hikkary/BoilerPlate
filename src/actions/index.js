import * as search from './search'
import * as counter from './counter'
import * as themes from './themes'

export default { counter, search, themes }
