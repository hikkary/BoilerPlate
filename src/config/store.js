import { applyMiddleware, createStore } from 'redux'
import logger from 'redux-logger'
// used for async func use on redux
// import thunk from 'redux-thunk'

import reducers from '../reducers'

const store = createStore(reducers, applyMiddleware(logger))

export default store
