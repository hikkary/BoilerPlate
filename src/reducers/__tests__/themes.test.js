import reducer from '../themes'

import { themeLight, themeDark } from '../../config/themes'
import { changeTheme } from '../../actions/themes'

const initialState = {
  currentTheme: themeLight
}

it('sets initial state', () => {
  const expected = { ...initialState }
  const actual = reducer(undefined, {})
  expect(actual).toEqual(expected)
})

it('set the light Theme  ', () => {
  const expected = {
    ...initialState,
    currentTheme: {
      ...themeLight
    }
  }

  const actual = reducer(undefined, changeTheme(themeLight))
  expect(actual).toEqual(expected)
})

// TODO: set the darklight

// MATCH SNAPSHOT

it('sets initial state Snapshot', () => {
  expect(reducer(undefined, {})).toMatchSnapshot()
  // reducer prend en premier argument un state, en second argument une action
})

// TODO: Snapshot quand on set themeDark et themeLight
