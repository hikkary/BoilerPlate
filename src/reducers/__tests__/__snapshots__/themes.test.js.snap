// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets initial state Snapshot 1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "backgroundColor": "#fff",
      "black": "#222",
      "bookSeparator": "#580303",
      "gray": "#979797",
      "gryffindorPrimary": "#7f0909",
      "gryffindorSecondary": "#d3a625",
      "hufflepuffPrimary": "#ecb939",
      "hufflepuffSecondary": "#372e29",
      "loginBackground": "#f2debd",
      "primaryColor": "#7f0909",
      "ravenclawPrimary": "#0e1a40",
      "ravenclawSecondary": "#946b2d",
      "secondaryColor": "#940B0B",
      "slytherinPrimary": "#1a472a",
      "slytherinSecondary": "#aaaaaa",
      "tertiaryColor": "#CB4B4B",
      "white": "#fff",
    },
  },
}
`;

exports[`sets themeDark Snapshot 1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "backgroundColor": "#fff",
      "black": "#222",
      "bookSeparator": "#580303",
      "gray": "#979797",
      "gryffindorPrimary": "#7f0909",
      "gryffindorSecondary": "#d3a625",
      "hufflepuffPrimary": "#ecb939",
      "hufflepuffSecondary": "#372e29",
      "loginBackground": "#f2debd",
      "primaryColor": "#7f0909",
      "ravenclawPrimary": "#0e1a40",
      "ravenclawSecondary": "#946b2d",
      "secondaryColor": "#940B0B",
      "slytherinPrimary": "#1a472a",
      "slytherinSecondary": "#aaaaaa",
      "tertiaryColor": "#CB4B4B",
      "white": "#fff",
    },
  },
}
`;

exports[`sets themeLight Snapshot 1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "backgroundColor": "#fff",
      "black": "#222",
      "bookSeparator": "#580303",
      "gray": "#979797",
      "gryffindorPrimary": "#7f0909",
      "gryffindorSecondary": "#d3a625",
      "hufflepuffPrimary": "#ecb939",
      "hufflepuffSecondary": "#372e29",
      "loginBackground": "#f2debd",
      "primaryColor": "#7f0909",
      "ravenclawPrimary": "#0e1a40",
      "ravenclawSecondary": "#946b2d",
      "secondaryColor": "#940B0B",
      "slytherinPrimary": "#1a472a",
      "slytherinSecondary": "#aaaaaa",
      "tertiaryColor": "#CB4B4B",
      "white": "#fff",
    },
  },
}
`;
